### PR TITLE
Feat: unlock forms

### LINKED ISSUE #1995: OpenGov. Add new unlock form screen
- Add new form screens for all types of wallets according to the design [Figma link](https://www.figma.com/design/ylvl7Ox25UuZj73Rn85DK7/OpenGov-voting?node-id=1864-112622&t=3yrMfX6OjAxXj9Q5-0)
- Add validation for the screens

### PR DESCRIPTION
**Pull request by sokolova-an**
- Added a new step for the unlock flow
- Update confirm screens

image

### CODE REVIEW

**Review by johnthecat — APPROVED** [review]

File: src/renderer/features/governance/aggregates/unlock.ts [hunk 1]
```diff
@@ -28,22 +30,28 @@ sample({
...
-      c.filter((claim) => claim.type === UnlockChunkType.CLAIMABLE),
-    ),
-    totalUnlock: unlockModel.$totalUnlock,
-  },
-  filter: ({ totalUnlock }) => !totalUnlock.isZero(),
-  fn: ({ unlockableClaims, totalUnlock }) => ({
-    event: { unlockableClaims, amount: totalUnlock.toString() },
+  clock: unlockFormStarted,
+  source: unlockModel.$claimSchedule,
+  fn: (claims) => claims.filter((claim) => claim.type === UnlockChunkType.CLAIMABLE) as ClaimChunkWithAddress[],
```
Comment (johnthecat) [inline src/renderer/features/governance/aggregates/unlock.ts:47]: This case is covered in typescript 5.5, 
```suggestion
  fn: (claims) => claims.filter((claim) => claim.type === UnlockChunkType.CLAIMABLE),
```

File: src/renderer/features/governance/components/Unlock/Confirmation.tsx
Comment (Asmadek) [inline src/renderer/features/governance/components/Unlock/Confirmation.tsx:1]: maybe move confirmation screen and model to OperationsConfirm? Or add a comment why it's here

File: src/renderer/features/governance/aggregates/unlockForm.ts [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+            return form.shards.every((_: AccountWithClaim, index: number) => {
+              return new BN(fee).lte(new BN(accounts[index]?.balance));
+            });
+          },
+        },
+      ],
+    },
+    description: {
+      init: '',
+      rules: [UnlockRules.description.maxLength],
```
Comment (Asmadek) [inline src/renderer/features/governance/aggregates/unlockForm.ts:163]: you can use more rules from `UnlockRules`, best way if all rules will be stored only in one place

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/governance/components/Unlock/Confirmation.tsx
Comment (sokolova-an) [reply src/renderer/features/governance/components/Unlock/Confirmation.tsx:1]: We want to apply the DDD approach for governance so trying to keep everything related to it in one domain (governance folder)

File: src/renderer/features/governance/aggregates/unlock.ts (hunk 1 above)
Comment (sokolova-an) [reply src/renderer/features/governance/aggregates/unlock.ts:47]: Hm, without this -  types don't match and it still believes that I'm trying to pass UnlockChunk[] not ClaimChunkWithAddress[]

**Comment by pgolovkin** [comment]
@sokolova-an missing key on the unlock formscreen
image

**Comment by pgolovkin** [comment]
@sokolova-an unlockable b

Uploading Screen Recording 2024-07-29 at 16.06.51.mov…

alance doesn't filled into the form (please, check the video).

**Comment by pgolovkin** [comment]
@sokolova-an unlockable balance doesn't filled into the form (please, check the video).


https://github.com/user-attachments/assets/4d5bedaa-4ba1-4501-9ed1-12be3883ecad

**Pull request merged**